### ISSUE TITLE
[Feature request] Possibility to allow only once a certain permission

### ISSUE THREAD

**Issue by k3tju** [GrapheneOS/os_issue_tracker#112]
Currently it is possible to deny once a certain permission for an application but allowing it once does not seem possible.

**Comment by thestinger** [comment]
This would be something compelling to provide but it needs to be a robust implementation. It would be best to enable for permissions case-by-case where it makes sense.

**Comment by thestinger** [comment]
It could also offer a timed permission grant.

**Comment by k3tju** [comment]
More info: There is the "Don't Ask again" box but, unfortunately, even if you do not tick it the permission is permanently granted.

**Comment by lazerl0rd** [comment]
Maybe this could use similar code to Android 10's "only once" for location?

**Comment by thestinger** [comment]
This is already implemented in Android 11 so we'll just wait for the standard implementation.

**Comment by thestinger** [comment]
Closing in favour of waiting for https://developer.android.com/preview/privacy/permissions#one-time.
